### PR TITLE
feat(outbound): add weighted random proxy-group selection

### PR DESCRIPTION
**Pull request by KagurazakaNyaa**
### Background

This change adds weighted random selection to proxy groups without introducing a new policy name.

Previously, `policy: random` always selected uniformly from alive nodes. This made it impossible to bias random selection toward preferred nodes while still keeping random behavior. This patch adds `add_weight` as a filter annotation, similar to the existing `add_latency` annotation, so users can write configurations like:

```dae
filter: name(US_node) [add_weight: 4]
policy: random
```

The change also fixes related edge cases discovered during implementation and review:

- zero-valued first annotations such as `add_latency: 0s` and `add_weight: 0`
- overflow around `1 + add_weight`
- aggregate overflow when selecting from groups with very large effective weights

In addition, it updates config marshal round-trip behavior so annotated `filter:` lines are preserved, and documents the actual first-match filter annotation semantics in the config description and example config.

Finally, it optimizes the weighted-random runtime path:

- equal-weight groups now use an O(1) fast path again
- normal weighted groups use integer roulette selection when safe
- only extreme overflow cases fall back to the exponential-race path

### Checklist

- [x] The Pull Request has been fully tested
- [x] There's an entry in the CHANGELOGS
- [x] There is a user-facing docs PR against https://github.com/daeuniverse/dae

### Full Changelogs

- Implement weighted random proxy-group selection via `add_weight` filter annotations
- Preserve annotated `filter:` lines during config marshal / round-trip
- Document first-match filter annotation semantics and weighted random usage in config docs and example config
- Fix zero-valued first-annotation handling and weight overflow edge cases
- Optimize weighted random selection with equal-weight fast path and safe integer weighted selection
- Add regression tests and performance benchmarks for weighted random selection

### Issue Reference

N/A

### Test Result

```bash
go test ./component/outbound/dialer -run 'TestNewAnnotation_|TestNewAliveDialerSet_AddWeightAcceptsLargestSafeEffectiveWeight|TestAliveDialerSet_GetRand_WithLargeAggregateWeights'

go test ./component/outbound -run 'TestDialerGroup_Select_Random|TestDialerGroup_SetAlive|TestDialerGroup_Select_Random_WithAddWeight|TestDialerGroup_Select_Random_WithAddWeight_IgnoreDeadDialer|TestDialerGroup_Select_Random_WithLargeAddWeight'

go test ./config -run 'TestMarshal'
```

Output:
```
ok      github.com/daeuniverse/dae/component/outbound/dialer    (cached)
ok      github.com/daeuniverse/dae/component/outbound   (cached)
ok      github.com/daeuniverse/dae/config       (cached)
```

Benchmark coverage added:

```bash
go test -run '^$' -bench '^BenchmarkAliveDialerSet_GetRand$' -benchmem -cpu=1 ./component/outbound/dialer
perf stat -e cycles -- /tmp/alive_dialer_bench.test ...
```

Measured scenarios include:

- `default_weights`
- `partial_add_weight`
- sizes: `10 / 100 / 1000 / 10000 / 100000`
- comparison against old `fastrand`-style uniform selection baseline

Output:
```
goos: linux
goarch: amd64
pkg: github.com/daeuniverse/dae/component/outbound/dialer
cpu: AMD Ryzen 7 7840HS w/ Radeon 780M Graphics
BenchmarkAliveDialerSet_GetRand/default_weights/weighted/10             100000000               10.84 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/default_weights/fastrand/10             100000000               10.38 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/default_weights/weighted/100            100000000               10.30 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/default_weights/fastrand/100            100000000               10.45 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/default_weights/weighted/1000           100000000               10.29 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/default_weights/fastrand/1000           100000000               10.32 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/default_weights/weighted/10000          112218357               10.88 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/default_weights/fastrand/10000          108281468               10.77 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/default_weights/weighted/100000         100000000               10.93 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/default_weights/fastrand/100000         100000000               10.96 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/weighted/10          11104996               106.4 ns/op             0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/fastrand/10          100000000               10.35 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/weighted/100          1260328               946.1 ns/op             0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/fastrand/100         100000000               10.45 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/weighted/1000          116056             10180 ns/op               0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/fastrand/1000        100000000               10.37 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/weighted/10000          10000            109467 ns/op               0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/fastrand/10000       111713031               10.61 ns/op            0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/weighted/100000           780           1593360 ns/op               0 B/op          0 allocs/op
BenchmarkAliveDialerSet_GetRand/partial_add_weight/fastrand/100000      100000000               11.16 ns/op            0 B/op          0 allocs/op
PASS
ok      github.com/daeuniverse/dae/component/outbound/dialer    26.385s
PASS

 Performance counter stats for '/tmp/alive_dialer_bench.test ../..':

         7,493,714      cycles

       0.002839727 seconds time elapsed

       0.001719000 seconds user
       0.001741000 seconds sys
```

Under the original configuration, there was no performance degradation. Under the weighted random configuration, the corresponding group's node selection time for establishing new inbound connections degraded from O(1) to O(n). However, for a scale of no more than a few thousand nodes with dozens of inbound connections per second, the performance impact was not significant.

### CODE REVIEW

File: component/outbound/dialer/alive_dialer_set_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	dialers := make([]*Dialer, size)
+	annotations := make([]*Annotation, size)
+	for i := range size {
+		dialers[i] = &Dialer{}
+		annotation := &Annotation{}
+		if weighted && i%10 == 0 {
+			annotation.AddWeight = 9
+		}
+		annotations[i] = annotation
+	}
```
[the remark this answers is not in the document]
Comment (KagurazakaNyaa) [reply component/outbound/dialer/alive_dialer_set_bench_test.go:28]: > `for i := range size` does not compile because `range` cannot iterate over an `int`. Use an index loop (e.g., `for i := 0; i < size; i++`) or range over the slice (e.g., `for i := range dialers`) to initialize `dialers`/`annotations`.

Thanks. This repository targets Go 1.24, and `for i := range size` is valid syntax since Go 1.22. I also verified it by compiling/running the benchmark successfully.
